tests/test__collection_api.py

```diff
@@ -5611,6 +5607,56 @@ def test__aggregate_array_to_object(self):
             with self.assertRaises(mongomock.OperationFailure):
                 collection.aggregate(item)
 
+    def test_aggregate_project_with_boolean(self):
```

Comment:
Please also add tests in `test__mongomock` to make sure the behavior is consistent with pymongo + MongoDB